kubemarine/core/connections.py

```diff
@@ -44,9 +44,13 @@ def get_connection(self, ip: str) -> fabric.connection.Connection:
     def _create_connection_from_details(self, ip: str, conn_details: dict,
                                         gateway: fabric.connection.Connection = None,
                                         inline_ssh_env: bool = True) -> fabric.connection.Connection:
-        # fabric / invoke use default encoding of deployer.
-        # We need to use encoding of remote nodes to correctly encode output of remote commands.
-        # Currently remote nodes are always Linux, so hard-code 'utf-8'.
+
+        creds={}
+        if conn_details.get('keyfile'):
+            creds['key_filename'] = os.path.expanduser(conn_details['keyfile'])
+        elif conn_details.get('password'):
+            creds['password'] = conn_details.get('password')
+        #else:     
```

Comment:
If user specified `keyfile` in `node_defaults` and `password` in `nodes` for specific node, kubemarine takes `keyfile` value for ssh connection and ignores node's `password`.
If it's OK, please mention it in IN guide.

And remove this debug `else`